src/Microsoft.AspNetCore.SignalR.Client/InvocationSubject.cs

```diff
@@ -0,0 +1,123 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.AspNetCore.SignalR.Client
+{
+    internal class InvocationSubject : IObservable<object>
+    {
```

Comment:
Yep, we should capture the observers into a new local list and call them outside the lock